Detections/Okta_CL/FailedLoginsFromUnknownOrInvalidUser.yaml

```diff
@@ -0,0 +1,34 @@
+id: 884be6e7-e568-418e-9c12-89229865ffde
+name: Failed Logins from Unknown or Invalid User
+description: |
+  'This creates an incident in the event that numerous login attempts to the management console with an unknown or invalid user name'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: OktaSSO
+    dataTypes: 
+      - Okta_CL
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - CredentialAccess 
+relevantTechniques:
+  - T1110
+query: |
+
+  let timeframe = ago(1h);
+  let FailureThreshold = 15;
+  let FailedLogins = Okta_CL
+  | where TimeGenerated > timeframe
+  | where eventType_s =~ "user.session.start" and outcome_reason_s =~ "UNKNOWN_USER"
+  | summarize count() by client_ipAddress_s, bin(TimeGenerated, 5m)
+  | where count_ > FailureThreshold
+  | project client_ipAddress_s;
+  Okta_CL
+  | where published_t > timeframe
+  | join kind=inner FailedLogins () on client_ipAddress_s
+  | where eventType_s =~ "user.session.start" and outcome_reason_s =~ "UNKNOWN_USER"
+  | summarize count() by ClientIP = client_ipAddress_s, City = client_geographicalContext_city_s, Country = client_geographicalContext_country_s, bin(TimeGenerated, 5m)
+  | sort by TimeGenerated desc
+  | extend timestamp = bin_TimeGenerated, IPCustomEntity = ClientIP
```

Comment:
There have been some optimizations on the backend so that you don't have to reference bin_TimeGenerated any longer and looks like it will fail now.  You can use timestamp = TimeGenerated or if you want a label then in the summarize label it bin_Time = bin(TimeGenerated, 5m) or something like that.

Reply:
Updated with timestamp = TimeGenerated